.github/pr_management/common.ts

```diff
@@ -0,0 +1,218 @@
+        })
+        .then(res => {
+            core.info("these comments were retrieved\n" + res);
+            return res;
+        })
+        .catch(err => {
+            throw err;
+        });
+
+    return comments.data.length > 0;
+}
+
+function isValidTimestamp(sinceTimeStamp: string) {
+    try {
+        Date.parse(sinceTimeStamp);
+    } catch (err) {
+        throw new Error(`the sinceTimeStamp argument passed is an invalid timestamp`);
+    }
+}
+
+//// abstractions for adding and dropping labels
+
+export async function addOngoingLabel() {
+    await addLabel("s.Ongoing");
+}
+
+export async function addToReviewLabel() {
+    await addLabel("s.ToReview");
+}
+
+export async function dropOngoingLabelAndAddToReview() {
+    await removeLabel("s.Ongoing");
+    await addLabel("s.ToReview");
+}
+
+export async function dropToReviewLabelAndAddOngoing() {
+    await removeLabel("s.ToReview");
+    await addLabel("s.Ongoing");
+}
+
+export async function dropOngoingLabel() {
+    await removeLabel("s.Ongoing");
+}
+
+async function addLabel(labelName: string) {
+    await octokit.rest.issues
+        .addLabels({
+            owner,
+            repo,
+            issue_number,
+            labels: [labelName],
+        })
+        .then(res => log.info(res.status, `added ${labelName} label with status`))
+        .catch(err => log.info(err, "error adding label"));
+}
+
+async function removeLabel(labelName: string) {
+    await octokit.rest.issues
+        .removeLabel({
+            owner,
+            repo,
+            issue_number,
+            name: labelName, // todo check if this works
+        })
+        .then(res => logInfo(res.status, `removing label ${res.status} with status`))
+        .catch(err => logInfo(err, "error removing label (label may not have been applied)"));
+}
+
+export async function sleep(ms : number) {
+    core.info(`sleeping for ${ms} milliseconds...`);
+    return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+////// functions to help with logging //////
+/* these functions log using the core module but with the format "label: itemToLog". 
+They also return the variable being logged for convenience */
+export const log = { info: logInfo, warn: logWarn, jsonInfo: jsonInfo };
+
+function logInfo(toPrint, label) {
+    core.info(`${label}: ${toPrint}`);
+    return toPrint;
+}
+
+function jsonInfo(jsonToPrint: JSON, label) {
+    core.info(`${label}: ${JSON.stringify(jsonToPrint)}`);
+}
+
+function logWarn(toPrint, label) {
+    core.warning(`${label}: ${toPrint}`);
+}
+
+//// comments
+export async function postComment(message) {
+    const commentBody = `Hi ${actor}, please note the following. ${message}`;
+
+    await octokit.rest.issues.createComment({
+        owner,
+        repo,
+        issue_number,
+        body: commentBody,
+    })
+    .then(res => core.info(`Commented:\n ${res.data.body}\n with status ${res.status}`))
+    .catch(err => core.error(err))
+}
+
+//// checks stuff
```

Comment:
I think this comment might not be informative. I also notice that u have `isValidTimestamp` and `wereReviewCommentsAdded` above, which also do checkings, should we place them here instead?